pkg/azclient/auth.go

```diff
@@ -68,7 +68,9 @@ func NewAuthProvider(
 	case len(config.AADMSIDataPlaneIdentityPath) > 0:
 		return newAuthProviderWithUserAssignedIdentity(config, clientOption, opts)
 	default:
```

Comment:
could you add a log for this scenario?

Reply:
Will add the log (when compute / network credential both nil) in the cloud-provider when upgrading the pkg in there. The azclient doesn't enable logging.  cc @MartinForReal

Reply:
Sure, that makes more sense!